docs/en/integrations/data-ingestion/apache-spark/spark-native-connector.md

```diff
@@ -0,0 +1,583 @@
+
+  df.write
+    .format("jdbc")
+    .mode("append")
+    .option("url", jdbcUrl)
+    .option("dbtable", "example_table")
+    .option("user", "default")
+    .option("password", "123456")
+    .option("SaveMode", "append")
+    .save()
+
+
+  // Stop the Spark session// Stop the Spark session
+  spark.stop()
+
+}
+```
+
+Find more details on the [`s3` table function page](/docs/en/sql-reference/table-functions/s3.md).
+
+</TabItem>
+<TabItem value="Python" label="Python">
+
+```python
+from pyspark.sql import SparkSession
+from pyspark.sql import Row
+
+jar_files = [
+    "jars/clickhouse-jdbc-X.X.X-SNAPSHOT-all.jar"
+]
+
+# Initialize Spark session with JARs
+spark = SparkSession.builder \
+    .appName("example") \
+    .master("local") \
+    .config("spark.jars", ",".join(jar_files)) \
+    .getOrCreate()
+
+# Create DataFrame
+data = [Row(id=11, name="John"), Row(id=12, name="Doe")]
+df = spark.createDataFrame(data)
+
+url = "jdbc:ch://localhost:8123/default"
+user = "your_user" 
+password = "your_password"  
+driver = "com.clickhouse.jdbc.ClickHouseDriver"
+
+# Write DataFrame to ClickHouse
+df.write \
+    .format("jdbc") \
+    .option("driver", driver) \
+    .option("url", url) \
+    .option("user", user) \
+    .option("password", password) \
+    .option("dbtable", "example_table") \
+    .mode("append") \
+    .save()
+
+
+```
+
+</TabItem>
+<TabItem value="SparkSQL" label="SparkSQL">
+
+```sql
+   CREATE TEMPORARY VIEW jdbcTable
+           USING org.apache.spark.sql.jdbc
+           OPTIONS (
+                   url "jdbc:ch://localhost:8123/default", 
+                   dbtable "schema.tablename",
+                   user "username",
+                   password "password",
+                   driver "com.clickhouse.jdbc.ClickHouseDriver" 
+           );
+   -- resultTable could be created with df.createTempView or with SparkSQL
+   INSERT INTO TABLE jdbcTable
+                SELECT * FROM resultTable;
+                
+```
+
+</TabItem>
+</Tabs>
+
+## DDL Operations
+
+You can perform DDL operations on your ClickHouse instance using SparkSQL, with all changes immediately persisted in ClickHouse. 
+SparkSQL allows you to write queries exactly as you would in ClickHouse, 
+so you can directly execute commands such as CREATE TABLE, TRUNCATE, and more - without modification, for instance:
+
+```sql
+
+use clickhouse; 
+
+CREATE TABLE test_db.tbl_sql (
+  create_time TIMESTAMP NOT NULL,
+  m           INT       NOT NULL COMMENT 'part key',
+  id          BIGINT    NOT NULL COMMENT 'sort key',
+  value       STRING
+) USING ClickHouse
+PARTITIONED BY (m)
+TBLPROPERTIES (
+  engine = 'MergeTree()',
+  order_by = 'id',
+  settings.index_granularity = 8192
+);
+```
+
+The above examples demonstrate SparkSQL queries, which you can run within your application using any API—Java, Scala, PySpark, or shell.
+
+
+## Supported Data Types
```

Comment:
could you extend the docs with the following sections:
- configuration options 
- adjusting clickhouse settings (if possible, see https://github.com/ClickHouse/spark-clickhouse-connector/pull/367)
- logging